stdlib/importlib/metadata/_meta.pyi

```diff
@@ -8,15 +9,34 @@ class PackageMetadata(Protocol):
     def __contains__(self, item: str) -> bool: ...
     def __getitem__(self, key: str) -> str: ...
     def __iter__(self) -> Iterator[str]: ...
-    def get_all(self, name: str, failobj: _T = ...) -> list[Any] | _T: ...
     @property
     def json(self) -> dict[str, str | list[str]]: ...
+    if sys.version_info >= (3, 12):
+        @overload
+        def get(self, name: str, failobj: None = None) -> str | None: ...
+        @overload
+        def get(self, name: str, failobj: _T) -> _T | str: ...
+        @overload
+        def get_all(self, name: str, failobj: None = None) -> list[Any] | None: ...
+        @overload
+        def get_all(self, name: str, failobj: _T) -> list[Any] | _T: ...
+    else:
+        def get_all(self, name: str, failobj: _T = ...) -> list[Any] | _T: ...
 
-class SimplePath(Protocol):
-    def joinpath(self) -> SimplePath: ...
-    def parent(self) -> SimplePath: ...
-    def read_text(self) -> str: ...
-    # There was a bug in `SimplePath` definition in cpython, see #8451
-    #  Strictly speaking `__div__` was defined in 3.10, not __truediv__,
-    # but it should have always been `__truediv__`.
-    def __truediv__(self) -> SimplePath: ...
+if sys.version_info >= (3, 12):
+    class SimplePath(Protocol[_T]):
+        def joinpath(self) -> _T: ...
+        @property
+        def parent(self) -> _T: ...
+        def read_text(self) -> str: ...
+        def __truediv__(self, other: _T | str) -> _T: ...
```

Comment:
Hmm, this has a bad annotation in the CPython/`importlib_metadata` source code. If other is a `str`, what's the inferred return type meant to be? We should probably leave this as it is for now, but maybe we should open an issue over at `importlib_metadata` asking for clarification

Reply:
Fix: https://github.com/python/importlib_metadata/pull/469

Reply:
Thank you!